tests/web/test_responses.py

```diff
@@ -0,0 +1,106 @@
+# tests/web/test_responses.py
+import pytest
+from starlette.applications import Starlette
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.routing import Route
+from starlette.testclient import TestClient
+
+from ludic.web.responses import (
+    BaseElement,
+    LudicResponse,
+    extract_response_status_headers,
+    prepare_response,
+)
+
+
+# Dummy BaseElement subclass for testing
+class DummyElement(BaseElement):
+    def to_html(self) -> str:
+        return "<p>Hello World</p>"
+
+
+# Dummy handler functions
+def plain_handler() -> str:
+    return "Hello"
+
+
+def tuple_handler() -> tuple[str, int]:
+    return "Hello", 201
+
+
+def triple_handler() -> tuple[str, int, dict[str, str]]:
+    return "Hello", 202, {"x-header": "yes"}
+
+
+def invalid_tuple() -> tuple[str, int, dict[str, str], str]:
+    return "Invalid", 201, {"x": "a"}, "extra"
+
+
+# ---------- UNIT TESTS ---------- #
+
+
+def test_extract_response_status_headers_two_elements() -> None:
+    r, code, headers = extract_response_status_headers(("hello", 404))
+    assert r[0] == "hello"
+    assert code == 404
+    assert headers is None
+
+
+def test_extract_response_status_headers_three_elements() -> None:
+    r, code, headers = extract_response_status_headers(("hello", 200, {"x": "1"}))
```

Comment:
Since the variable is not used, you can do the following:

```suggestion
    _, code, headers = extract_response_status_headers(("hello", 200, {"x": "1"}))
```